core/finality-grandpa/src/tests.rs

```diff
@@ -306,6 +306,14 @@ impl ApiExt<Block> for RuntimeApi {
 	fn runtime_version_at(&self, _: &BlockId<Block>) -> Result<RuntimeVersion> {
 		unimplemented!("Not required for testing!")
 	}
+
+	fn record_proof(&mut self) {
+		unimplemented!("Not required for testing!")
```

Comment:
This seems a bit worrying: if these are not required for testing, then the code that calls them is not being tested.  That indicates a weakness in our test coverage.

Reply:
There is nothing to worry about, if that is not required for this test, it is not required and nothing to worry about. I need to implement the trait, but these functions are not required.